libminifi/src/controllers/SSLContextService.cpp

```diff
@@ -80,7 +82,16 @@ bool SSLContextService::configure_ssl_context(SSL_CTX *ctx) {
         X509_free(cert);
```

Comment:
My gut feeling says that whenever we do this we leak the page pointed by "ca". 
PKCS12_parse allocates that as well, so I think we need to free it. 

I would also prefer using lock_guards for the resources here to avoid leaks.

Reply:
Changed the code to use scope guards.